### PR TITLE
added UNION to split fall & spring semesters; added calc for retention

### PR DESCRIPTION
**Pull request by anthonygwalters**
# Pull Request

## Summary & Motivation

[//]: # (When merged, this pull request will...)

## Self-review

- [x] Update due date and assignee on the [Teamster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)
- [x] <kbd>Format</kbd> has been run on all modified files

## Troubleshooting
- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205983367053261

### CODE REVIEW

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__enrollment_pivot.sql
Comment (cbini) [inline src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__enrollment_pivot.sql:1]: This addition doesn't appear to be used anywhere

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__kfwd_persistence.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
+select
+    p.*,
```
Comment (cbini) [inline src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__kfwd_persistence.sql:2]: list out the columns

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql [hunk 2]
```diff
@@ -74,13 +152,64 @@ select
             e.actual_end_date is null and eis.ugrad_status in ('Graduated', 'Attending')
         then 1
         else 0
-    end as is_persisting_spring,
+    end as is_persisting_int,
+
+    case
+        when
+            date(r.academic_year + 1, 3, 31)
```
Comment (cbini) [inline src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql:159]: I think you can refactor this calc to the roster CTE and avoid having to repeat it throughout the rest of the SELECT

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql [hunk 3]
```diff
@@ -57,7 +48,94 @@ select
...
         else 0
-    end as is_persisting_fall,
+    end as is_persisting_int,
+
+    case
+        when date(r.academic_year, 10, 31) > current_date('{{ var("local_timezone") }}')
+        then null
+        when
+            e.actual_end_date >= date(r.academic_year, 10, 31)
+            and eis.ecc_account_id = e.school
```
Comment (anthonygwalters) [inline src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql:58]: @cbini this is where the addition is used

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__enrollment_pivot.sql
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__enrollment_pivot.sql:1]: see comment below

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql (hunk 3 above)
Comment (cbini) [reply src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql:58]: oh I see it now

File: src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql (hunk 2 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/kippadb/intermediate/int_kippadb__persistence.sql]: done

File: src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__kfwd_persistence.sql (hunk 1 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/extracts/tableau/rpt_tableau__kfwd_persistence.sql]: done

**Review by cbini — APPROVED** [review]

**Pull request merged**